Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -234,7 +235,34 @@ private async Task<bool> RefreshOidcTokenAsync(TokenRefreshManager caller, OidcT
             return true;
         }
 
-        private async Task<string> GetDefaultServiceAccountEmail()
+        /// <summary>
+        /// Signs the provided blob using the private key associated with the service account
+        /// this ComputeCredential represents.
+        /// </summary>
+        /// <param name="blob">The blob to sign.</param>
+        /// <param name="cancellationToken">Cancellation token to cancel the operation.</param>
+        /// <returns>The base64 encoded signature.</returns>
+        /// <exception cref="HttpRequestException">When the signing request fails.</exception>
+        /// <exception cref="JsonException">When the signing response is not valid JSON.</exception>
+        /// <remarks>
+        /// The private key associated with the Compute service account is not known locally
+        /// by a ComputeCredential. Signing happens by executing a request to the IAM Credentials API
+        /// which increases latency and counts towards IAM Credentials API quotas. Aditionally, the first
+        /// time a ComputeCredential is used to sign data, a request to the metadata server is execute to
+        /// to obtain the email of the default Compute service account.
+        /// </remarks>
+        public async Task<string> SignBlobAsync(byte[] blob, CancellationToken cancellationToken = default)
+        {
+            var request = new IamSignBlobRequest { Payload = blob };
+            var signBlobUrl = string.Format(GoogleAuthConsts.IamSignEndpointFormatString, await GetDefaultServiceAccountEmailAsync(cancellationToken).ConfigureAwait(false));
```

Comment:
I wonder whether for the sake of both readability and debugging it's worth separating this out:

```csharp
var request = new IamSignBlobRequest { Payload = blob };
var serviceAccountEmail = await GetDefaultServiceAccountEmailAsync(cancellationToken).ConfigureAwait(false);
var signBlobUrl = string.Format(GoogleAuthConsts.IamSignEndpointFormatString, serviceAccountEmail);
...
```

Reply:
Yep, will do that.